### PR TITLE
Improvement/migrate aes crypto lib

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

Migration to the current version of aes-crypto lib from the forked one.

### CODE REVIEW

File: app/core/Vault.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	const mnemonic = await KeyringController.exportSeedPhrase(password);
+	return JSON.stringify(mnemonic).replace(/"/g, '');
+};
+
+/**
+ * Recreates a vault
+ *
+ * @param password - Password to recreate and set the vault with
+ */
+export const recreateVault = async (password = '', selectedAddress) => {
```
Comment (estebanmino) [inline app/core/Vault.js:19]: is this taking imported accounts or it's going to be deleted?

**Review by ibrahimtaveras00 — APPROVED** [review]
looks good, QA Passed 👍

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Still fails on imported accounts as mentioned in https://github.com/MetaMask/metamask-mobile/pull/1756

File: app/core/Vault.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * Recreates a vault with the same password for the purpose of using the newest encryption methods
+ *
+ * @param password - Password to recreate and set the vault with
+ */
+export const recreateVaultWithSamePassword = async (password = '', selectedAddress) => {
+	const { KeyringController, PreferencesController } = Engine.context;
+	const seedPhrase = await getSeedPhrase(password);
+
+	let importedAccounts = [];
```
Comment (estebanmino) [inline app/core/Vault.js:23]: seems like is the same code right? https://github.com/MetaMask/metamask-mobile/pull/1756/files#diff-05500d00f4a6959745b57a220987a4b6R359 
would be nice to have `Logger` here to know what's going on and remove duplicated code, but we can do the last after releasing

**Comment by ibrahimtaveras00** [comment]
Imported accounts are not lost on login 👍 

I did notice that we show a different error message on android with empty password or wrong password rather than how in iOS we show Invalid password

seen here = https://recordit.co/K8r7qnxUl8

Screen Shot 2020-08-11 at 5 30 04 PM

Again this is only on android.  To reproduce:
- either sync with extension/import via seed phrase/ or create a new wallet
- after arriving on wallet view, open drawer and log out 
- now on login screen simply tap log in or input an invalid password

**Comment by andrepimenta** [comment]
@ibrahimtaveras00 just fixed the error message on Android 👍  You can take a look again

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍

**Pull request merged**